unified-runtime/source/loader/layers/sanitizer/msan/msan_shadow.cpp

```diff
@@ -344,38 +344,49 @@ ur_result_t MsanShadowMemoryGPU::AllocLocalShadow(ur_queue_handle_t Queue,
 }
 
 ur_result_t MsanShadowMemoryGPU::AllocPrivateShadow(ur_queue_handle_t Queue,
-                                                    uint32_t NumWG, uptr &Begin,
-                                                    uptr &End) {
-  const size_t RequiredShadowSize = NumWG * MSAN_PRIVATE_SIZE;
-  static size_t LastAllocedSize = 0;
-  if (RequiredShadowSize > LastAllocedSize) {
-    auto ContextInfo = getMsanInterceptor()->getContextInfo(Context);
-    if (PrivateShadowOffset) {
-      UR_CALL(getContext()->urDdiTable.USM.pfnFree(
-          Context, (void *)PrivateShadowOffset));
-      PrivateShadowOffset = 0;
-      LastAllocedSize = 0;
-    }
-
+                                                    uint64_t NumWI,
+                                                    uint32_t NumWG, uptr *&Base,
```

Comment:
why is Base a reference to a pointer? All the code seems to be doing with it is taking its address and then casting it to void**.

Reply:
Because the type of "Base" is "uptr *", that is, an array of "uptr"

Reply:
Right, but why do you need it to be a reference? It's an 8 byte value that you directly cast into a (void**). Begin and End is optionally modified as part of this function, Base isn't.

Anyway, not a big deal, just something that stood out to me ;) I'll resolve the conversation, feel free to change it or not.